### ISSUE TITLE
Docker build failure

### ISSUE THREAD

**Issue by dashengSun** [EOSIO/eos#1232]
The reason:
```
Step 18/26 : COPY --from=builder /usr/local/lib/* /usr/local/lib/
 ---> Using cache
 ---> e01b6497df4e
Step 19/26 : COPY --from=builder /opt/eosio/bin /opt/eosio/bin
COPY failed: stat /var/lib/docker/aufs/mnt/932731c8c65c2f41e6c8ffed3faeb96ce2d19a999b603979efe359e8687a375c/opt/eosio/bin: no such file or directory
```

The result of docker version:
```
Client:
 Version:       17.12.0-ce
 API version:   1.35
 Go version:    go1.9.2
 Git commit:    c97c6d6
 Built: Wed Dec 27 20:03:51 2017
 OS/Arch:       darwin/amd64

Server:
 Engine:
  Version:      17.12.0-ce
  API version:  1.35 (minimum version 1.12)
  Go version:   go1.9.2
  Git commit:   c97c6d6
  Built:        Wed Dec 27 20:12:29 2017
  OS/Arch:      linux/amd64
  Experimental: true
```
Does anyone else occur this issue?

**Comment by dallasjohnson** [comment]
I'm experiencing the same error. Have you made any progress with it?

**Comment by njir** [comment]
I think it's same with [issue#1146](https://github.com/EOSIO/eos/issues/1146)

**Comment by tusik** [comment]
same issue

**Comment by tusik** [comment]
@njir It's not work for me...

**Comment by Akk3d1s** [comment]
Having the same issue :(

**Comment by njir** [comment]
@tusik I removed all docker images for EOS and built again after changing some files like below. The build works fine for me...

```diff
diff --git a/Docker/Dockerfile b/Docker/Dockerfile
index 89ea2b22..d2f0973e 100644
--- a/Docker/Dockerfile
+++ b/Docker/Dockerfile
@@ -73,9 +73,9 @@ RUN git clone --depth 1 -b releases/stable git://github.com/mongodb/mongo-cxx-dr
 RUN git clone -b master --depth 1 https://github.com/EOSIO/eos.git --recursive \
     && cd eos \
     && cmake -H. -B"/tmp/build" -GNinja -DCMAKE_BUILD_TYPE=Release -DWASM_LLVM_CONFIG=/opt/wasm/bin/llvm-config -DCMAKE_CXX_COMPILER=clang++ \
-       -DCMAKE_C_COMPILER=clang -DCMAKE_INSTALL_PREFIX=/opt/eos  -DSecp256k1_ROOT_DIR=/usr/local \
-    && cmake --build /tmp/build --target install
-
+       -DCMAKE_C_COMPILER=clang -DCMAKE_INSTALL_PREFIX=/opt/eosio  -DSecp256k1_ROOT_DIR=/usr/local \
+    && cmake --build /tmp/build --target install \
+       && cd /tmp/build && cmake -DCMAKE_INSTALL_PREFIX=/opt/eosio -P cmake_install.cmake


 FROM ubuntu:16.04
diff --git a/Docker/config.ini b/Docker/config.ini
index 6401c549..11e51c14 100644
--- a/Docker/config.ini
+++ b/Docker/config.ini
@@ -1,6 +1,6 @@
 # File to read Genesis State from
 # genesis-json =
-genesis-json = "/opt/eos/bin/data-dir/genesis.json"
+genesis-json = "/opt/eosio/bin/data-dir/genesis.json"

 # the location of the block log (absolute path or relative to application data dir)
 block-log-dir = "blocks"
```

Then I run below script and it works fine.
```
$ docker run --name eosiod -p 8888:8888 -p 9876:9876 -t eosio/eos start_eosiod.sh --genesis-json /opt/eosio/bin/data-dir/genesis.json
```

**Comment by tusik** [comment]
@njir That work now, I don't know why still got "COPY failed"

**Comment by Akk3d1s** [comment]
@njir Thanks buddy, that did the trick :)

**Comment by tusik** [comment]
@njir but when i use `docker-compose up` still use path `/opt/eos/bin/data-dir/genesis.json`
how to change this
`eosiod_1   | fc::exists( my->genesis_file ): unable to find genesis file '/opt/eos/bin/data-dir/genesis.json', check --genesis-json argument
eosiod_1   |     {"f":"/opt/eos/bin/data-dir/genesis.json"}
eosiod_1   |     thread-0  chain_plugin.cpp:189 plugin_startup
eosiod_1   |
eosiod_1   |     {"my->genesis_file.generic_string()":"/opt/eos/bin/data-dir/genesis.json"}
eosiod_1   |     thread-0  chain_plugin.cpp:210 plugin_startup`

**Comment by njir** [comment]
if you want use ```docker-compose up```, modify ```docker-compose.yml```
```diff
       - "8888"
     volumes:
       - eosiod-data-volume:/opt/eosio/bin/data-dir
+      - ./config.ini:/opt/eosio/bin/data-dir/config.ini
```
it's just trick. I hope it updates soon

**Comment by dashengSun** [comment]
Still failing when running `docker build . -t eosio/eos` with modifying
```
-    && cmake --build /tmp/build --target install
+    && cmake --build /tmp/build --target install \
+    && cd /tmp/build && cmake -DCMAKE_INSTALL_PREFIX=/opt/eos -P cmake_install.cmake
```
and
```
-genesis-json = "/opt/eos/bin/data-dir/genesis.json"
+genesis-json = "/opt/eosio/bin/data-dir/genesis.json"
```

**Comment by Akk3d1s** [comment]
This is what I get when running the following:
$ docker run --name eosiod -p 8888:8888 -p 9876:9876 -t eosio/eos start_eosiod.sh arg1 arg2
**standard_init_linux.go:195: exec user process caused "no such file or directory"**

**Comment by Akk3d1s** [comment]
Anyone?

**Comment by ericiles** [comment]
This should be resolved with PR #1218
